### PR TITLE
Adding VSTS file

### PR DESCRIPTION
**Pull request by ryanbrandenburg**

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
Its seems like this has some real error happening, will need to investigate more.

**Comment by ryanbrandenburg** [comment]
This is stalled while we wait for a SME to have time to investigate getting NGINX installed on our agents. Just running install-nginx.sh doesn't work.

**Comment by ryanbrandenburg** [comment]
@muratg can we get a SME to look into this?

**Comment by muratg** [comment]
@ryanbrandenburg I'm not sure who has the experience in the team.

@mikeharder / @shirhatti  / @natemcmaster  / @anurse  do any of you guys have experience with nginx?

**Comment by anurse** [comment]
I have no background on our use of nginx here, but it looks like we're building it from source which seems fraught with peril. Could we just change the script to install from package managers or maybe use docker?

**Comment by anurse** [comment]
I have no background on our use of nginx here, but it looks like we're building it from source which seems fraught with peril. Could we just change the script to install from package managers or maybe use docker?

File: .vsts-pipelines/builds/ci-public.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+phases:
+- template: .vsts-pipelines/templates/phases/default-build.yml@buildtools
+  parameters:
+    agentOs: Windows
+
+- template: .vsts-pipelines/templates/phases/default-build.yml@buildtools
+  parameters:
+    agentOs: macOS
+    beforeBuild:
+    - script: brew install nginx
```
Comment (natemcmaster) [inline .vsts-pipelines/builds/ci-public.yml]: I assume we want these for the ci-internal builds too, right?

**Comment by ryanbrandenburg** [comment]
I fixed the NGINX install issues, but there's a failing test on linux that an area expert should look at.

**Comment by muratg** [comment]
@ryanbrandenburg Have a bug tracking the failing test?  Can you ping @Tratcher there?

**Comment by Tratcher** [comment]
Classes of failures:
- Travis linx build didn't even start?
- VSTS Windows & AppVeyor: ANCM V2 tests failing @jkotalik? I assume the schema is missing?
- VSTS Linux: 
`[0.414s] Microsoft.AspNetCore.Server.IntegrationTesting.NginxDeployer warning : nginx start stderr: nginx: [emerg] invalid number of arguments in "user"` @BrennanConroy ?

**Comment by jkotalik** [comment]
@Tratcher yeah the schema is missing. We can add a script to fix it on here and music store. We may want to centralize it to build tools.

File: .vsts-pipelines/builds/ci-public.yml (hunk 1 above)
Comment (ryanbrandenburg) [reply .vsts-pipelines/builds/ci-public.yml]: Done.

**Comment by ryanbrandenburg** [comment]
@Tratcher, @jkotalik was there any progress?

**Comment by jkotalik** [comment]
@ryanbrandenburg I added a few things to your PR. I'll debug appropriately.

**Comment by natemcmaster** [comment]
Adding myself to this PR to move this along.

**Comment by natemcmaster** [comment]
macOS passes.

Linux fails one test: `System.Exception : Failed to start nginx`

Windows fails to boot the build at all.
```
2018-07-26T18:00:37.9226598Z Generating script.
2018-07-26T18:00:37.9273570Z Script contents:
2018-07-26T18:00:37.9278834Z ..\..\tools\update_schema.ps1
2018-07-26T18:00:38.0106370Z ##[command]"C:\Windows\system32\cmd.exe" /D /E:ON /V:OFF /S /C "CALL "D:\a\_temp\2bb94e51-7e6f-4a25-98ca-870cb5cc9e6b.cmd""
2018-07-26T18:00:38.0219234Z The system cannot find the path specified.
2018-07-26T18:00:38.1891518Z ##[error]Cmd.exe exited with code '1'.
```

File: tools/update_schema.ps1 [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 1
+
+$ancmSchemaFiles = @(
+    "aspnetcore_schema.xml",
+    "aspnetcore_schema_v2.xml"
+)
+
+$ancmSchemaFileLocations = @(
+    @(Resolve-Path "$PSScriptRoot\..\src\AspNetCoreModuleV1\AspNetCore\aspnetcore_schema.xml"),
```
Comment (natemcmaster) [inline tools/update_schema.ps1]: @jkotalik I'm assuming these file paths are meant to be for IISIntegration, not the ServerTests repo, right? Any reason I shouldn't just copy that file into ServerTests?

Comment (jkotalik) [reply tools/update_schema.ps1]: Yeah let's go ahead and copy.

**Comment by natemcmaster** [comment]
:up: :date: 
it's working

**Comment by ryanbrandenburg** [comment]
:shipit: All hail Nate! Travis is failing, but it looks like it's specific to them, so we can deal with it in a different PR.

File: test/update_schema.ps1 [hunk 3]
```diff
@@ -0,0 +1,72 @@
+<#
```
Comment (jkotalik) [inline test/update_schema.ps1]: We should put this in build tools or part of korebuild eventually. Anywhere we use IIS tests will fail.

File: test/aspnetcore_schema.xml [hunk 4]
```diff
@@ -0,0 +1,39 @@
+<!--
```
Comment (jkotalik) [inline test/aspnetcore_schema.xml]: How easy would it be to get the schema file from the Microsoft.AspNetCore.AspNetCoreModuleV2 package? This solution isn't maintainable though we most likely won't update the schema any longer.

Comment (natemcmaster) [reply test/aspnetcore_schema.xml]: I think this should become part of the IIS test deployers, not KoreBuild. The deployer  could check for the updated schema, attempt to update it if it's out of date, and maybe fail the test if it doesn't have the right permissions.

**Pull request merged**